homework/HW2_queries.sql

```diff
@@ -0,0 +1,120 @@
+--- Connect to database
+\c starschema_company;
+
+--- Get the top 3 product types that have proven most profitable
+SELECT sum(o.profit) as sum_profit, p.product_line
+FROM order_line_fact_table AS o
+LEFT JOIN products as p
+ON o.product_code = p.product_code
+group by p.product_line
+order by sum_profit desc
+limit 3;
+
+
+--- Get the top 3 products by most items sold
+SELECT sum(o.quantity_ordered) as sum_ordered, p.product_name
+FROM order_line_fact_table AS o
+LEFT JOIN products as p
+ON o.product_code = p.product_code
+group by p.product_code
+order by sum_ordered desc
+limit 3;
+
+--- Get the top 3 products by items sold per country of customer for: USA, Spain, Belgium
+SELECT sum(op.quantity_ordered) as sum_ordered, op.product_name
+FROM (order_line_fact_table AS o
+LEFT JOIN products as p
+ON o.product_code = p.product_code) as op
+left join customers as c
+on op.customer_number=c.customer_number
+where c.country='USA'
+group by op.product_name
+order by sum_ordered desc
+limit 3;
+
+SELECT sum(op.quantity_ordered) as sum_ordered, op.product_name
+FROM (order_line_fact_table AS o
+LEFT JOIN products as p
+ON o.product_code = p.product_code) as op
+left join customers as c
+on op.customer_number=c.customer_number
+where c.country='Spain'
+group by op.product_name
+order by sum_ordered desc
+limit 3;
+
+SELECT sum(op.quantity_ordered) as sum_ordered, op.product_name
+FROM (order_line_fact_table AS o
+LEFT JOIN products as p
+ON o.product_code = p.product_code) as op
+left join customers as c
+on op.customer_number=c.customer_number
+where c.country='Belgium'
+group by op.product_name
+order by sum_ordered desc
+limit 3;
+
+--- Get the most profitable day of the week
+SELECT sum(o.profit) as sum_profit, d.day_of_the_week
+FROM order_line_fact_table AS o
+LEFT JOIN dates as d
+ON o.date_key = d.date_key
+group by d.day_of_the_week
+order by sum_profit desc
+limit 1;
+
+--- Get the top 3 city-quarters with the highest average profit margin in their sales
+SELECT sum(oi.profit)/sum(oi.sales_amount) as avg_profit_margin, oi.quarter,e.city
+FROM (order_line_fact_table AS o
+LEFT JOIN dates as d
+ON o.date_key = d.date_key) as oi
+left join employees as e
+on oi.sales_rep_employee_number=e.employee_number
+group by oi.quarter,e.city
+order by avg_profit_margin desc
+limit 3;
+
+-- List the employees who have sold more goods (in $ amount) than the average employee.
+
+
+/*select sum(o.sales_amount) as sum_sales, e.employee_number,e.first_name,e.last_name
```

Comment:
Delete commented-out code before committing! Keep it in old commits if you want to return to it.